### PR TITLE
Add quiet and output to job monitor to suppress or redirect status messages

### LINKED ISSUE #1978: Add quiet= and to_err= to job monitor to suppress or redirect status messages.
### Add quiet= and to_err= to job monitor to suppress or redirect status messages. 

If one captures output from a script, `qiskit.tools.monitor.job_monitor` (and its factor, `_text_checker`) spew extraneous info. Suppress with `quiet=True` or redirect to stderr with `to_err=True`.

### PR DESCRIPTION
**Pull request by jwoehr**
### Summary

Fixes #1978

This PR adds 2 new kwargs to the job monitor, `quiet` and `output`, to suppress or redirect the status messages generated by the job monitor.

### Details and comments

This is a continuation from #1979

### CODE REVIEW

File: qiskit/tools/monitor/job_monitor.py [hunk 1]
```diff
@@ -52,19 +57,22 @@ def _text_checker(job, interval, _interval_set=False):
...
+        if msg != prev_msg and not quiet:
+            print('\r%s: %s' % ('Job Status', msg), end='', file=_outstream)
             prev_msg = msg
-    print('')
+    if not quiet:
+        print('', file=_outstream)
 
 
-def job_monitor(job, interval=None, monitor_async=False):
+def job_monitor(job, interval=None, monitor_async=False, quiet=False, to_file=None):
```
Comment (mtreinish) [inline qiskit/tools/monitor/job_monitor.py]:
```suggestion
def job_monitor(job, interval=None, monitor_async=False, quiet=False, output=sys.stdout):
```

File: qiskit/tools/monitor/job_monitor.py [hunk 2]
```diff
@@ -52,19 +57,22 @@ def _text_checker(job, interval, _interval_set=False):
...
     """Monitor the status of a IBMQJob instance.
 
     Args:
         job (BaseJob): Job to monitor.
         interval (int): Time interval between status queries.
         monitor_async (bool): Monitor asyncronously (in Jupyter only).
+        quiet (bool): If True, do not print status messages.
+        to_file (file): If file print status messages to it, else to stdout.
```
Comment (mtreinish) [inline qiskit/tools/monitor/job_monitor.py]:
```suggestion
        output (file): The file like object to write status messages to. By default this is sys.stdout.
```

File: qiskit/tools/monitor/job_monitor.py [hunk 3]
```diff
@@ -85,17 +93,20 @@ def job_monitor(job, interval=None, monitor_async=False):
...
             status = widgets.HTML(value=header % job.status().value)
             display(status)
 
             thread = threading.Thread(target=_html_checker, args=(job, interval,
                                                                   status, header))
             thread.start()
         else:
-            _text_checker(job, interval, _interval_set)
+            _text_checker(job, interval, _interval_set,
+                          quiet=quiet, to_file=to_file)
```
Comment (mtreinish) [inline qiskit/tools/monitor/job_monitor.py]:
```suggestion
                          quiet=quiet, output=output)
```

File: qiskit/tools/monitor/job_monitor.py [hunk 4]
```diff
@@ -85,17 +93,20 @@ def job_monitor(job, interval=None, monitor_async=False):
...
 
     else:
         if monitor_async:
-            raise QiskitError('monitor_async only available in Jupyter notebooks.')
-        _text_checker(job, interval, _interval_set)
+            raise QiskitError(
+                'monitor_async only available in Jupyter notebooks.')
+        _text_checker(job, interval, _interval_set, quiet=quiet, to_file=to_file)
```
Comment (mtreinish) [inline qiskit/tools/monitor/job_monitor.py]:
```suggestion
        _text_checker(job, interval, _interval_set, quiet=quiet, output=output)
```

File: qiskit/tools/monitor/job_monitor.py [hunk 5]
```diff
@@ -19,20 +19,25 @@
     from IPython.display import display    # pylint: disable=import-error
 
 
-def _text_checker(job, interval, _interval_set=False):
+def _text_checker(job, interval, _interval_set=False, quiet=False, to_file=None):
```
Comment (mtreinish) [inline qiskit/tools/monitor/job_monitor.py]:
```suggestion
def _text_checker(job, interval, _interval_set=False, quiet=False, output=sys.stdout):
```

File: qiskit/tools/monitor/job_monitor.py [hunk 6]
```diff
@@ -19,20 +19,25 @@
...
 
     Args:
         job (BaseJob): The job to check.
         interval (int): The interval at which to check.
         _interval_set (bool): Was interval time set by user?
+        quiet (bool): If True, do not print status messages.
+        to_file (file): If file print status messages to it, else to stdout.
+
     """
+    _outstream = to_file if to_file else sys.stdout
```
Comment (mtreinish) [inline qiskit/tools/monitor/job_monitor.py]:
```suggestion
```

File: qiskit/tools/monitor/job_monitor.py [hunk 7]
```diff
@@ -19,20 +19,25 @@
...
     status = job.status()
     msg = status.value
     prev_msg = msg
     msg_len = len(msg)
 
-    print('\r%s: %s' % ('Job Status', msg), end='')
+    if not quiet:
+        print('\r%s: %s' % ('Job Status', msg), end='', file=_outstream)
```
Comment (mtreinish) [inline qiskit/tools/monitor/job_monitor.py]:
```suggestion
        print('\r%s: %s' % ('Job Status', msg), end='', file=output)
```

File: qiskit/tools/monitor/job_monitor.py [hunk 8]
```diff
@@ -52,19 +57,22 @@ def _text_checker(job, interval, _interval_set=False):
         elif len(msg) > msg_len:
             msg_len = len(msg)
 
-        if msg != prev_msg:
-            print('\r%s: %s' % ('Job Status', msg), end='')
+        if msg != prev_msg and not quiet:
+            print('\r%s: %s' % ('Job Status', msg), end='', file=_outstream)
```
Comment (mtreinish) [inline qiskit/tools/monitor/job_monitor.py]:
```suggestion
            print('\r%s: %s' % ('Job Status', msg), end='', file=output)
```

File: qiskit/tools/monitor/job_monitor.py [hunk 9]
```diff
@@ -52,19 +57,22 @@ def _text_checker(job, interval, _interval_set=False):
...
             prev_msg = msg
-    print('')
+    if not quiet:
+        print('', file=_outstream)
```
Comment (mtreinish) [inline qiskit/tools/monitor/job_monitor.py]:
```suggestion
        print('', file=output)
```

File: qiskit/tools/monitor/job_monitor.py [hunk 10]
```diff
@@ -19,20 +19,25 @@
...
-def _text_checker(job, interval, _interval_set=False):
+def _text_checker(job, interval, _interval_set=False, quiet=False, to_file=None):
     """A text-based job status checker
 
     Args:
         job (BaseJob): The job to check.
         interval (int): The interval at which to check.
         _interval_set (bool): Was interval time set by user?
+        quiet (bool): If True, do not print status messages.
+        to_file (file): If file print status messages to it, else to stdout.
```
Comment (mtreinish) [inline qiskit/tools/monitor/job_monitor.py]:
```suggestion
        output (file): The file like object to write status messages to. By default this is sys.stdout.
```

File: qiskit/tools/monitor/job_monitor.py (hunk 1 above)
Comment (jwoehr) [reply qiskit/tools/monitor/job_monitor.py]: Will make these changes today-ish, tomorrow-ish ...

Comment (jwoehr) [reply qiskit/tools/monitor/job_monitor.py]: And yes, @mtreinish , I misunderstood the Github workflow and made the same pr-close-open mistake on a couple of submissions. I _think_ I have it straight now :)

**Comment by jwoehr** [comment]
Thanks, merged all suggestions @mtreinish

**Comment by jwoehr** [comment]
Oops. should have batched those commits of your suggestions ... still learning the interface, @mtreinish :)

**Pull request merged**